src/post/workshed_utils.ts

```diff
@@ -67,6 +67,26 @@ export function getBestCycle(): TrainSet.Cycle {
   return [...trainCycle];
 }
 
+export function getBestOffset(): number {
+  const cycle = getBestCycle();
+  const valueTrainStation = (station: TrainSet.Station) =>
+    POTENTIAL_BEST_TRAIN_PIECES.find(({ piece }) => piece === station)?.value?.() ?? 0;
+  const cycleValue =
+    5 * (sum(POTENTIAL_BEST_TRAIN_PIECES, ({ value }) => value()) + valueTrainStation(cycle[0]));
+
+  const potentialOffsets = Array(POTENTIAL_BEST_TRAIN_PIECES.length)
+    .fill(null)
+    .map((_x, i) => i + 2);
+
+  return maxBy(potentialOffsets, (offset) => {
+    const tail = [...cycle].splice(1, offset - 1);
```

Comment:
Is `slice` more efficient here, since we don't need to make unnecessary `[...cycle]` copies - (although we should note that it's `slice(idx, idx+delta)` as compared to `splice(idx, delta)`)